client/src/common/components/Filter.jsx

```diff
@@ -22,14 +25,30 @@ class Filter extends PureComponent {
     this.debouncedFilter.cancel();
   }
 
-  filter = query => {
-    const { options, onFilter } = this.props;
+  filterByFields = query => {
+    const { fields, options } = this.props;
+    const normalizedQuery = query.toLowerCase();
+
+    return options.filter(option => {
+      let testString = '';
+
+      fields.forEach(field => {
+        if (option[field]) {
+          testString += option[field];
+        }
+      });
+
+      return testString.toLowerCase().includes(normalizedQuery);
+    });
+  };
+
+  handleFilter = query => {
```

Comment:
`handleFilterChange`? or what does this method do exactly

Reply:
name changed